kernel: bump 5.15 to 5.15.116

graysky2 opened the pull request:
All patches rebased automatically.
    
    Build system: x86_64
    Build-tested: ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
    Run-tested: ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

graysky2 commented:
I am getting a build error with ramips/tplink_archer-a6-v3:
```
make[8]: Entering directory '/scratch/union/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/linux-5.15.115'
  CC      drivers/net/dsa/mt7530.o
drivers/net/dsa/mt7530.c: In function 'mt7531_setup':
drivers/net/dsa/mt7530.c:2453:14: error: 'i' undeclared (first use in this function); did you mean 'id'?
 2453 |         for (i = 0; i < MT7530_NUM_PORTS; i++)
      |              ^
      |              id
drivers/net/dsa/mt7530.c:2453:14: note: each undeclared identifier is reported only once for each function it appears in
make[8]: *** [scripts/Makefile.build:289: drivers/net/dsa/mt7530.o] Error 1
```

DragonBluep commented:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?h=v5.15.115&id=5a7266feaa6d708fc6880a161786eaa884ef3c8e
In this patch, upstream version removes the variable `i`, but our local version still retains it.

graysky2 commented:
@DragonBluep - I don't understand, our local version was removed in this commit.

DragonBluep commented:
What I mean is the upstream merged patch is not exactly the same as our local patch so we can't fully remove it. There is a tiny difference about variable `i`. We need to add a new patch to redefine variable `i`.

upstream version:
```
 mt7531_setup(struct dsa_switch *ds)
 {
 	struct mt7530_priv *priv = ds->priv;
 	struct mt7530_dummy_poll p;
-	struct dsa_port *cpu_dp;
 	u32 val, id;
-	int ret, i;
+	int ret;
 ```
local version:
https://github.com/openwrt/openwrt/blob/0a4b309f41062ef40706162ae53b6428982a0685/target/linux/generic/backport-5.15/790-v6.4-0010-net-dsa-mt7530-split-off-common-parts-from-mt7531_se.patch#L78-L85

graysky2 commented:
Not sure why the PR is closed now/I cannot reopen it.

graysky2 committed:
kernel: bump 5.15 to 5.15.115

Manually rebased:
	backport-5.15/603-v5.19-page_pool-Add-recycle-stats-to-page_pool_put_page_bu.patch
	pending-5.15/723-net-mt7531-ensure-all-MACs-are-powered-down-before-r.patch*

Removed upstreamed:
	generic-backport/610-v6.3-net-page_pool-use-in_softirq-instead.patch[1]
	backport-5.15/705-12-v6.0-net-dsa-mt7530-rework-mt753-01-_setup.patch[2]
	backport-5.15/790-v6.4-0010-net-dsa-mt7530-split-off-common-parts-from-mt7531_se.patch[3]
	backport-5.15/703-10-v5.16-net-dsa-introduce-helpers-for-iterating-through-port.patch[4]

All other patches automatically rebased.

* Modified to define the variable i as suggested by DragonBluep in PR discussion.
  See: https://github.com/openwrt/openwrt/pull/12823#issuecomment-1578518576

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.115&id=3af319d5147454dc63665ef451229c674b538377
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.115&id=0753c1ef24194580f7165ae6e259b59a851392f2
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v5.15.115&id=5a7266feaa6d708fc6880a161786eaa884ef3c8e
4. https://github.com/gregkh/linux/commit/9902f91cf666124a6b50bbcf483b46ecb09ef408

Build system: x86_64
Build-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: bcm2711/RPi4B, ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
Thanks for the suggestions.  Built fine and seems to be running fine on `ramips/tplink_archer-a6-v3` for me.  Can you please review?

I was a bit confused because there are two sections that could fit the lines starting with `/* Reset whole chip through gpio pin or memory-mapped registers for` ... I placed the block in the one closest to the definition for `i` as you can see.

DragonBluep commented:
Tested on ath79, everything is fine. Most of the changes are related to the mt7531 switch. If mt7621 runs well, there's nothing to worry about.

Borromini commented:
I've got a full day of homeworking, so it looks like MT7621 is okay. Was tested on top of 23.05 HEAD but shouldn't matter much yet.

Compile-tested: ath79/generic, octeon, ramips/mt76{20,21,x8}, realtek/rtl{838,930}x, mvebu/cortexa{9,72}
Run-tested: mt7621 (EAP615-Wall, WAX202), mvebu (RB5009UG+S+IN)

CharlesMengCA commented:
Linksys E8450 (UBI) Tested.  Working fine.

hauke commented on target/linux/generic/backport-5.15/703-10-v5.16-net-dsa-introduce-helpers-for-iterating-through-port.patch:
```diff
@@ -31,9 +31,9 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
- }
+@@ -504,6 +504,34 @@ static inline bool dsa_is_user_port(stru
+ 	dsa_switch_for_each_port((_dp), (_ds)) \
+ 		if (dsa_port_is_cpu((_dp)))
```
This patch is not needed any more. It was upstreamed here: https://github.com/gregkh/linux/commit/9902f91cf666124a6b50bbcf483b46ecb09ef408

graysky2 replied:
I removed it, thank you.  I did not re-do the update since 5.15.116 is out now... doing it there.  Please let me know if that is unacceptable.

graysky2 committed:
kernel: bump 5.15 to 5.15.116

All patches rebased automatically.

Build system: x86_64
Build-tested: ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod
Run-tested: ramips/tplink_archer-a6-v3, filogic/xiaomi_redmi-router-ax6000-ubootmod

Signed-off-by: John Audia <therealgraysky@proton.me>

graysky2 commented:
@hauke - not sure why the error
https://github.com/openwrt/openwrt/actions/runs/5231935142/jobs/9446653258?pr=12823#step:29:32

This is already set:
https://github.com/openwrt/openwrt/blob/main/target/linux/generic/config-5.15#L1636

Ansuel commented:
@graysky2 maybe some kernel modules to update?

graysky2 commented:
@Ansuel - Might need some help with that.  Not sure I ever did that before.

DragonBluep commented:
> @hauke - not sure why the error https://github.com/openwrt/openwrt/actions/runs/5231935142/jobs/9446653258?pr=12823#step:29:32

The missing symbol is `CONFIG_DRM_RCAR_USE_LVDS`
> This is already set: https://github.com/openwrt/openwrt/blob/main/target/linux/generic/config-5.15#L1636
`CONFIG_DRM_RCAR_LVDS`

The existing one is `CONFIG_DRM_RCAR_LVDS`

They are different!

zxlhhyccc commented:
> @hauke- 不知道为什么错误 https://github.com/openwrt/openwrt/actions/runs/5231935142/jobs/9446653258?pr=12823#step:29:32
> 
> 这已经设置好了： https://github.com/openwrt/openwrt/blob/main/target/linux/generic/config-5.15#L1636

`DRM_RCAR_USE_LVDS`
`CONFIG_DRM_RCAR_LVDS`

graysky2 committed:
kernel: add CONFIG_DRM_RCAR_USE_LVDS is not set

Added missing symbol.

Signed-off-by: John Audia <therealgraysky@proton.me>
